tests/integ/test_workflow_with_fail_steps.py

```diff
@@ -0,0 +1,325 @@
+            pass
+
+
+def test_two_step_fail_pipeline_with_no_err_msg(sagemaker_session, role, pipeline_name):
+    param = ParameterInteger(name="MyInt", default_value=2)
+    cond = ConditionEquals(left=param, right=1)
+    step_fail = FailStep(
+        name="FailStep",
+    )
+    step_cond = ConditionStep(
+        name="CondStep",
+        conditions=[cond],
+        if_steps=[],
+        else_steps=[step_fail],
+    )
+    pipeline = Pipeline(
+        name=pipeline_name,
+        steps=[step_cond],
+        sagemaker_session=sagemaker_session,
+        parameters=[param],
+    )
+
+    try:
+        response = pipeline.create(role)
+        pipeline_arn = response["PipelineArn"]
+        execution = pipeline.start(parameters={})
+        response = execution.describe()
+        assert response["PipelineArn"] == pipeline_arn
+
+        try:
+            execution.wait(delay=30, max_attempts=60)
+        except WaiterError:
+            pass
+        execution_steps = execution.list_steps()
+
+        assert len(execution_steps) == 2
+        for execution_step in execution_steps:
+            if execution_step["StepName"] == "CondStep":
+                assert execution_step["StepStatus"] == "Succeeded"
+                continue
+            assert execution_step["StepName"] == "FailStep"
+            assert execution_step["StepStatus"] == "Failed"
+            assert execution_step.get("FailureReason", None) is None
+            metadata = execution_steps[0]["Metadata"]["Fail"]
+            assert metadata["ErrorMessage"] == ""
+
+        # Check FailureReason field in ListPipelineExecutions
+        executions = sagemaker_session.sagemaker_client.list_pipeline_executions(
+            PipelineName=pipeline.name
+        )["PipelineExecutionSummaries"]
+
+        assert len(executions) == 1
+        assert executions[0]["PipelineExecutionStatus"] == "Failed"
+        assert (
+            "Step failure: One or multiple steps failed"
+            in executions[0]["PipelineExecutionFailureReason"]
+        )
+    finally:
+        try:
+            pipeline.delete()
+        except Exception:
+            pass
+
+
+def test_invalid_pipeline_depended_on_fail_step(sagemaker_session, role, pipeline_name):
+    param = ParameterInteger(name="MyInt", default_value=2)
+    cond = ConditionEquals(left=param, right=1)
+    step_fail = FailStep(
+        name="FailStep",
+        error_message="Failed pipeline execution",
+    )
+    step_cond = ConditionStep(
+        name="CondStep",
+        conditions=[cond],
+        if_steps=[],
+        else_steps=[],
+        depends_on=["FailStep"],
+    )
+    pipeline = Pipeline(
+        name=pipeline_name,
+        steps=[step_cond, step_fail],
+        sagemaker_session=sagemaker_session,
+        parameters=[param],
+    )
+
+    try:
+        with pytest.raises(Exception) as error:
+            pipeline.create(role)
+
+        assert "CondStep can not depends on FailStep" in str(error.value)
```

Comment:
This error message is generated from the SM Pipeline backend. The truth is FailStep can not be depended on/ referenced by any other steps since FailStep will terminate the pipeline execution. 

There is a grammar error in this sentence but we have to update the backend before we can make any changes here. Otherwise, it will break the integration test.
As this is just a test rather than source code, we can add it into follow up items and update later.